### ISSUE TITLE
GraphQL operations definition with yml does not work

### ISSUE THREAD

**Issue by klobastov** [api-platform/core#5318]
**API Platform version(s) affected**: 3.0.8

**Description**  
GraphQL documents are not generated.

Also, when I remove the use of the configuration mapping `"%kernel.project_dir%/config/api_platform"`, then GraphQL documents are generated successfully.

But when I try to use entity annotation, then they are ignored. This way I can describe either an openapi schema for the API in yml or describe operations for GraphQL in php.

![image](https://user-images.githubusercontent.com/3277751/211765542-6dcdc845-ff6e-407a-aae2-b9a39a22614f.png)

**How to reproduce**  
When I try to perform a graphql operation, I get an error 
`ApiPlatform\Metadata\Resource\Factory\LinkFactory::createLinksFromRelations(): Argument #1 ($operation) must be of type ApiPlatform\Metadata\ApiResource|ApiPlatform\Metadata\Operation, array given, called in /app/vendor/api-platform/core/src/Metadata/Resource/Factory/LinkResourceMetadataCollectionFactory.php on line 43`

config/api_platform/resources/entity/gate.yaml
```
    graphQlOperations:
      queries:
        store_house_list:
          class: ApiPlatform\Metadata\GraphQl\QueryCollection
```

**Additional Context**  
config/packages/api_platform.yaml
```
  mapping:
    paths:
      - '%kernel.project_dir%/src/Gate/Entity'
      - '%kernel.project_dir%/src/ApiResource/Poco'
      - '%kernel.project_dir%/config/api_platform'
```

config/api_platform/resources/entity/gate.yaml
```
resources:
  Gate\Entity\Gate:
    order:
      serialId: ASC

    normalizationContext:
      groups: [ 'gate:output' ]
    denormalizationContext:
      groups: [ 'gate:input' ]

    operations:
      gate_list:
        class: 'ApiPlatform\Metadata\GetCollection'
        filters: [ 'default.id', 'default.name' ]
        security: is_granted('GATE_LIST', object)

      gate_view:
        class: 'ApiPlatform\Metadata\Get'
        security: is_granted('GATE_LIST', object)

      gate_create:
        class: 'ApiPlatform\Metadata\Post'
        security: is_granted('GATE_CREATE', object)

      gate_update:
        class: 'ApiPlatform\Metadata\Put'
        security: is_granted('GATE_UPDATE', object) and (previous_object.getCreatedBy() === user.getUsername() or is_my_storehouse(object.getStoreHouse().getId() ?? null))

      gate_update_partial:
        class: 'ApiPlatform\Metadata\Patch'
        security: is_granted('GATE_UPDATE', object) and (previous_object.getCreatedBy() === user.getUsername() or is_my_storehouse(object.getStoreHouse().getId() ?? null))

      gate_delete:
        class: 'ApiPlatform\Metadata\Delete'
        security: is_granted('GATE_CREATE', object) and (previous_object.getCreatedBy() === user.getUsername() or is_my_storehouse(object.getStoreHouse().getId() ?? null))
```
Enity:
```
#[ORM\Entity(repositoryClass: GateRepository::class)]
#[ApiResource]
class Gate
{
    use BlameableEntity;
    use TimestampableEntity;
    use SequencedEntity;

    #[ORM\Id]
    #[ORM\Column(type: UuidType::NAME, unique: true)]
    #[ORM\GeneratedValue(strategy: 'CUSTOM')]
    #[ORM\CustomIdGenerator(class: 'doctrine.uuid_generator')]
    private ?Uuid $id = null;
...
}
```

Generated API resources
![image](https://user-images.githubusercontent.com/3277751/211765352-4b1b8046-24e2-422f-b37c-91a72ff0c4bb.png)
And Json Schema
![image](https://user-images.githubusercontent.com/3277751/211769311-1595564c-8d5a-40af-a857-4a309048a180.png)

**Comment by OndrosI** [comment]
i have the same problem. I went back to 3.0.7 and it works just fine

**Comment by klobastov** [comment]
Solved after update to 3.0.9 (#5315) or rollback to  3.0.7

**Comment by aharabara** [comment]
I will leave this for a stranger who tries to figure out how to solve this error (assuming it is Symfony context):

**SOLUTION:** Go to `api_platform.yaml` and set `api_platform.graphql.enabled` to `true`.
 
![image](https://github.com/api-platform/core/assets/31312666/bb488886-9250-45dd-b147-b8abd2bf0ecd)
